process/transaction/shardProcess.go

```diff
@@ -451,62 +453,39 @@ func (txProc *txProcessor) processMoveBalance(
 
 func (txProc *txProcessor) processSCDeployment(
 	tx *transaction.Transaction,
-	adrSrc []byte,
+	acntSrc state.UserAccountHandler,
 ) (vmcommon.ReturnCode, error) {
-	// getAccounts returns acntSrc not nil if the adrSrc is in the node shard, the same, acntDst will be not nil
-	// if adrDst is in the node shard. If an error occurs it will be signaled in err variable.
-	acntSrc, err := txProc.getAccountFromAddress(adrSrc)
-	if err != nil {
-		return 0, err
-	}
-
 	return txProc.scProcessor.DeploySmartContract(tx, acntSrc)
 }
 
 func (txProc *txProcessor) processSCInvoking(
 	tx *transaction.Transaction,
-	adrSrc, adrDst []byte,
+	acntSrc, acntDst state.UserAccountHandler,
 ) (vmcommon.ReturnCode, error) {
-	// getAccounts returns acntSrc not nil if the adrSrc is in the node shard, the same, acntDst will be not nil
-	// if adrDst is in the node shard. If an error occurs it will be signaled in err variable.
-	acntSrc, acntDst, err := txProc.getAccounts(adrSrc, adrDst)
-	if err != nil {
-		return 0, err
-	}
-
 	return txProc.scProcessor.ExecuteSmartContractTransaction(tx, acntSrc, acntDst)
 }
 
 func (txProc *txProcessor) processBuiltInFunctionCall(
 	tx *transaction.Transaction,
-	adrSrc, adrDst []byte,
+	acntSrc, acntDst state.UserAccountHandler,
 ) (vmcommon.ReturnCode, error) {
-	// getAccounts returns acntSrc not nil if the adrSrc is in the node shard, the same, acntDst will be not nil
-	// if adrDst is in the node shard. If an error occurs it will be signaled in err variable.
-	acntSrc, acntDst, err := txProc.getAccounts(adrSrc, adrDst)
-	if err != nil {
-		return 0, err
-	}
-
 	return txProc.scProcessor.ExecuteBuiltInFunction(tx, acntSrc, acntDst)
 }
 
 func (txProc *txProcessor) processRelayedTx(
 	tx *transaction.Transaction,
-	adrSrc, adrDst []byte,
+	relayerAcnt, acntDst state.UserAccountHandler,
 ) (vmcommon.ReturnCode, error) {
 
 	_, args, err := txProc.argsParser.ParseCallData(string(tx.GetData()))
 	if err != nil {
-		return 0, err
-	}
-
-	relayerAcnt, acntDst, err := txProc.getAccounts(adrSrc, adrDst)
-	if err != nil {
+		//TODO: check if return code 0 is ok here
```

Comment:
yes it is. otherwise this is a critical error - these transactions were already verified by interceptor.

Reply:
ok